lib/Dialect/FIRRTL/Ops.cpp

```diff
@@ -74,6 +75,118 @@ static LogicalResult verifyCircuitOp(CircuitOp &circuit) {
     return failure();
   }
 
+  // Store a mapping of defname strings to the first external module
+  // that defines it.
+  llvm::StringMap<FExtModuleOp> defnameMap;
+
+  // Verify external modules.
+  for (auto &op : *circuit.getBody()) {
+    if (auto extModule = dyn_cast<FExtModuleOp>(op)) {
+      if (auto defname = extModule.defnameAttr()) {
```

Comment:
I'd recommend utilizing `continue` a bit more here to reduce nesting, e.g.:
```
    auto extModule = dyn_cast<FExtModuleOp>(op);
    if (!extModule) continue;
```

Reply:
Good suggestion. Flattened that to just the outer loop.